packages/retail-ui-extensions/src/extension-api/scanner-api/scanner-api.ts

```diff
@@ -0,0 +1,24 @@
+import type {RemoteSubscribable} from '@remote-ui/async-subscription';
+
+/** The scanner source the POS device supports */
+export type SupportedSource = 'camera' | 'external' | 'embedded';
+
+export interface ScannerSubscriptionResult {
+  /** The string data from the last scanner event received */
+  data?: string;
+  /** An array containing all supported scanner types on the POS device. */
+  supportedSources: SupportedSource[];
```

Comment:
This one should be the source of the specific scan. So where the data comes from.

Reply:
We can add this to the actual API object, this array shouldn't mutate, right? Or will it actually show if an external scanner is connected? If it does, then perhaps we should just have a separate subscribable for this?

Reply:
supported sources is an array that returns the available scanners that device has available. It is not related to the type of scanner data that gets returned. Hmm maybe bunching this array into the same hook is causing confusion and we should have a separate hook for it.

Reply:
I agree. It's fine here. I think ConnectedSources or AvailableSources might be a less confusing term than SupportedSources, but it's not a strong opinion.